src/runaqa.ts

```diff
@@ -1,324 +1,475 @@
+  }
+  if (jdksource.length !== 0) {
+    parameters += ` --sdk_resource ${jdksource}`
+  }
+  if (customizedSdkUrl.length !== 0) {
+    parameters += ` --customizedURL ${customizedSdkUrl}`
+  }
+  if (sdkdir.length !== 0) {
+    parameters += ` --sdkdir ${sdkdir}`
+  }
+  if (IS_WINDOWS) {
+    await exec.exec(`bash ./get.sh ${parameters} ${vendorTestParams}`)
+  } else {
+    await exec.exec(`./get.sh ${parameters} ${vendorTestParams}`)
+  }
+}
+
+/**
+ * Sets up enviroment to generate parallelList.mk
+ * @param  {string} version JDK Version being tested
+ * @param  {string} jdksource Source for JDK
+ * @param  {[string]} customizedSdkUrl Download link for JDK binaries
+ * @param  {[string]} sdkdir Directory for SDK
+ * @param  {[string]} buildList AQAvit Test suite
+ * @param  {[string]} aqatestsRepo Alternative aqatestRepo
+ * @param  {[string]} openj9Repo Alternative openj9Repo
+ * @param  {[string]} tkgRepo Alternative TKG repo
+ * @param  {[string]} vendorTestParams Vendor provided test parameters
+ * @param  {[string]} aqasystemtestsRepo Alternative AQA-systemtestRepo
+ * @return {[null]}  null
+ */
+export async function setupParallelEnv(
+  version: string,
+  jdksource: string,
+  customizedSdkUrl: string,
+  sdkdir: string,
+  buildList: string,
+  aqatestsRepo: string,
+  openj9Repo: string,
+  tkgRepo: string,
+  vendorTestParams: string,
+  aqasystemtestsRepo: string,
+  numMachines: string
+): Promise<void> {
+
+  await setupTestEnv(version, jdksource, customizedSdkUrl, sdkdir, buildList, aqatestsRepo, openj9Repo, tkgRepo, vendorTestParams, aqasystemtestsRepo);
+  process.chdir('TKG');
+  process.env.PARALLEL_OPTIONS = `PARALLEL_OPTIONS=TEST=${buildList} TEST_TIME= NUM_MACHINES=${numMachines}`;
+  await exec.exec(`make genParallelList ${process.env.PARALLEL_OPTIONS}`);
+
+}
+
+/**
+ * Sets up the test environment on the runner.
+ * @param  {string} version JDK Version being tested
+ * @param  {string} jdksource Source for JDK
+ * @param  {[string]} customizedSdkUrl Download link for JDK binaries
+ * @param  {[string]} sdkdir Directory for SDK
+ * @param  {[string]} buildList AQAvit Test suite
+ * @param  {[string]} aqatestsRepo Alternative aqatestRepo
+ * @param  {[string]} openj9Repo Alternative openj9Repo
+ * @param  {[string]} tkgRepo Alternative TKG repo
+ * @param  {[string]} vendorTestParams Vendor provided test parameters
+ * @param  {[string]} aqasystemtestsRepo Alternative AQA-systemtestRepo
+ * @return {null}  null
+ */
+async function setupTestEnv(
+  version: string,
+  jdksource: string,
+  customizedSdkUrl: string,
+  sdkdir: string,
+  buildList: string,
+  aqatestsRepo: string,
+  openj9Repo: string,
+  tkgRepo: string,
+  vendorTestParams: string,
+  aqasystemtestsRepo: string
+  ):  Promise<void>{
+    await installPlatformDependencies();
+    setSpec();
+    process.env.BUILD_LIST = buildList;
+    if ((jdksource === 'upstream' ||

```

Comment:
Could extract this into a small private method?

Reply:
Refactored.